### ISSUE TITLE
Template props

### ISSUE THREAD

**Issue by Mat-Moo** [vuejs/Discussion#306]
I'm trying to hack into the bootstrap modal window with some success, however I'm sending 2 optional props for it, a title and footer. The title is fine, however in the footer I want to send 

```
<moo-modal
  v-if='modal'
  footer='<button class="btn bt-primary" v-on="click:test()">Close</button>'...
```
template code
```
    template: '<div class="modal fade in" style="display:block;">' +
        '<div class="modal-dialog"><div class="modal-content">' +
        '<div v-if="title" class="modal-header"><div class="panel-title"><h4 class="modal-title">{{title}}</h4></div></div>' +
        '<div class="modal-body"><content></content></div>' +
        '{{$data|json}}' +
        '<div v-if="footer" class="modal-footer">{{{footer}}}</div>' +
        '</div></div></div><div class="modal-backdrop fade in"></div>'
```
So when modal is true it creates the component and shows it, the footer shows a button but no event is there, just the text v-on=... I need to compile that bit of code on created almost (tried $eval etc.)
Or is this a limitation of templates hence the new slot system, and would that fix this? Or am I doing it ham-fisted?

**Comment by swift1** [comment]
I think the way you're doing it is flawed. The html template should be in the template, not in the actual props you're passing in. Think about what differentiates the different footers from each other. Maybe the only difference is the footer text. If so, you should only pass the footer text in as footer props and put the rest in the template. If there are more differences, like what link to click on when you click on the different footers, then just pass the name of the function as a footer prop.

**Comment by Mat-Moo** [comment]
The footers are all different, with various buttons for each modal window, so no not flawed at all. If I wanted all the footers the same then yes it would be in the template, that is common sense. How would you do the footer then to have different buttons for different windows?

In terms of mess yes it is, but I'll be converting to https://github.com/vuejs/vue-component-compiler anway, thanks for the hint though, at least I can see the divs are all correct :)

**Comment by swift1** [comment]
It's still flawed, cause you could just put all the different footers in templates and pass the template name as a prop instead. Anyway, it would be easier to help you if you provided a link to a fiddle.

**Comment by Mat-Moo** [comment]
Wouldn't be much of a template then would it?

http://jsbin.com/yafago/edit?html,js,output

**Comment by swift1** [comment]
Working example (click on [Modal 5 with title and footer + button] to test the click handler):

http://jsfiddle.net/hajkrupo/7/

**Comment by Mat-Moo** [comment]
Thank you I appreciate the effort but seems like a dirty way to go around it. Be interested to hear other peoples opinions.

**Comment by swift1** [comment]
Yes, there's definitely still room for improvement on that code, for example by putting all the modals in separate templates instead of using ````v-if````.

**Comment by young-steveo** [comment]
@Mat-Moo I agree with @swift1 .  Putting template strings in component properties seems like a misuse of the feature.  His solution is a much cleaner way to achieve what you're looking to do.

**Comment by Mat-Moo** [comment]
Yes because of the limits of component templates, which is why https://github.com/vuejs/Discussion/issues/298 makes a lot of sense

**Comment by young-steveo** [comment]
@Mat-Moo If you don't need the footer component flexibility, you can also do this: http://jsfiddle.net/b9ke8uuy/3/

**Comment by Mat-Moo** [comment]
Thanks, but I want to be able to pass different buttons to emit different events, like cancel/ok/help, for now I can add the footer to my content - I probably won't even use bootstrap anyway, so this is all a learning exercise :+1: Just trying to make things modular...
That said still keen to see how others have approached this

**Comment by swift1** [comment]
I took the time to rewrite the whole code, and something like this is how I think it should be done:

http://jsfiddle.net/hajkrupo/9/
